src/clj/rems/application/search.clj

```diff
@@ -0,0 +1,69 @@
+(ns rems.application.search
+  (:require [clojure.java.io :as io]
+            [clojure.tools.logging :as log]
+            [mount.core :as mount]
+            [rems.config :refer [env]]
+            [rems.db.applications :as applications]
+            [rems.db.events :as events])
+  (:import [org.apache.lucene.analysis.standard StandardAnalyzer]
+           [org.apache.lucene.document Document StringField Field$Store TextField]
+           [org.apache.lucene.index IndexWriter IndexWriterConfig IndexWriterConfig$OpenMode]
+           [org.apache.lucene.queryparser.flexible.standard StandardQueryParser]
+           [org.apache.lucene.search IndexSearcher ScoreDoc TopDocs SearcherManager SearcherFactory]
+           [org.apache.lucene.store Directory NIOFSDirectory]))
+
+(def ^:private analyzer (StandardAnalyzer.))
+
+(mount/defstate ^Directory search-index
+  :start (let [directory (NIOFSDirectory. (.toPath (io/file (:search-index-path env))))]
```

Comment:
should you guard against a a nil :search-index-path here?

Reply:
There is a default for it in config-defaults.edn, so it should never be nil.